doc/src/manual/style-guide.md

```diff
@@ -164,6 +164,65 @@ uses (e.g. `a[i]::Int`) than to try to pack many alternatives into one type.
 If a function name requires multiple words, consider whether it might represent more than one
 concept and might be better split into pieces.
 
+## Write functions that use similar argument precedence as Julia's `base/`
+
+As a general rule, the Base library uses the following order of arguments to functions,
+as applicable:
+
+1. **Function argument**.
+   This permits the use of [`do`](@ref) blocks for passing multiline anonymous functions.
+
+2. **I/O stream**.
+   Specifying the `IO` object first permits passing the function to functions such as
+   [`sprint`](@ref), e.g. `sprint(show, x)`.
+
+3. **Input being mutated**.
+   In [`fill!(x, v)`](@ref fill!), `x` is the object being mutated, and it appears before
+   the value to be inserted into `x`.
+
+4. **Type**.
+   Passing a type typically means that the output will have the given type.
+   In [`parse(Int, "1")`](@ref parse), the type comes before the string to parse.
+   There are many such examples where the type appears first, but it's useful to note that
+   in [`read(io, String)`](@ref read), the `IO` argument appears before the type, which is
```

Comment:
Are discussions about changing this order resolved?

Reply:
Not really: @JeffBezanson is against changing it, I'm in favor of changing it. Others fall in various places on that spectrum.

Reply:
I can just choose a different example if you know of a better one that isn't being reconsidered.

Reply:
I don't think so, because that's the whole debate. ;-)

Reply:
Resolved on triage.